lib/Master.php

```diff
@@ -80,13 +87,63 @@ public function __construct(GlobalSiteSelector $gss,
 	 * @param array $param
 	 */
 	public function handleLoginRequest($param) {
-		$uid = $param['uid'];
-		$password = $param['password'];
 
-		$location = $this->queryLookupServer($uid);
+		// if there is a valid JWT it is a internal GSS request between master and slave
+		// -> skip login
```

Comment:
what is then the purpose if the request (=== why was it fired?)

Reply:
then, why is the pre_login hook being fired on the autologout endpoint?

Reply:
because we sending a request to the global site selector and are not logged in at this server. So it tries various auth options, including apache auth before it continues to the public page. There the pre_login hook is fired